### PR TITLE
FUSETOOLS-2748 - Factorize code to retrieve online property values

### PR DESCRIPTION
**Pull request by apupier**
Signed-off-by: Aurélien Pupier <apupier@redhat.com>

# Pull Request Checklist

After this checklist is all checked or PR provides explanations for possible pass-through, please put the label "Ready for review" on the PR.

## General

- [x] Did you use the Jira Issue number in the commit comments?
- [x] Did you set meaningful commit comments on each commit?
- [x] Did you sign off all commits for this PR? (git commit -s -m "jira issue number - your commit comment")

## Functional

- [x] Did the CI job report a successful build?
- [x] Is the non-happy path working, too?

## Maintainability

- [ ] Are all Sonar reported issues fixed or can they be ignored?
- [x] Is there no duplicated code?
- [ ] Are method-/class-/variable-names meaningful?

## Tests

- [x] Are there unit-tests?
- [ ] Are there integration tests (or at least a jira to tackle these)?
- [ ] Do we need a new UI test?

## Legal

- [x] Have you used the correct file header copyright comment?
- [x] Have you used the correct year in the headers of new files?

### CODE REVIEW

File: core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/util/CamelCatalogUtils.java [hunk 1]
```diff
@@ -41,74 +44,18 @@
...
 	public static final String CAMEL_SPRING_BOOT_STARTER = "camel-spring-boot-starter";
 	public static final String CAMEL_WILDFLY = "org.wildfly.camel";
 	
 	public static final String CAMEL_VERSION_LATEST_COMMUNITY = "2.20.2";
-	public static final String CAMEL_VERSION_LATEST_PRODUCTIZED_62 = FUSE_621_R9_CAMEL_VERSION;
-	public static final String CAMEL_VERSION_LATEST_PRODUCTIZED_63 = FUSE_63_R5_CAMEL_VERSION;
-	public static final String CAMEL_VERSION_LATEST_FIS_20 = FIS_20_R3_CAMEL_VERSION;
+	public static final String CAMEL_VERSION_LATEST_PRODUCTIZED_62 = CamelForFuse6ToBomMapper.FUSE_621_R9_CAMEL_VERSION;
+	public static final String CAMEL_VERSION_LATEST_PRODUCTIZED_63 = CamelForFuse6ToBomMapper.FUSE_63_R5_CAMEL_VERSION;
+	public static final String CAMEL_VERSION_LATEST_FIS_20 = CamelForFIS6ToBomMapper.FIS_20_R3_CAMEL_VERSION;
```
Comment (lhein) [inline core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/util/CamelCatalogUtils.java]: Isn't it FIS 2.0 ? Why did u choose FIS6 here?

File: core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/util/CamelCatalogUtils.java [hunk 2]
```diff
@@ -41,74 +44,18 @@
...
-	private static final String FUSE_63_R3_BOM_VERSION = "6.3.0.redhat-262";
-	static final String FUSE_63_R4_BOM_VERSION = "6.3.0.redhat-283";
-	static final String FUSE_63_R5_BOM_VERSION = "6.3.0.redhat-310";
-	
-	public static final String FIS_20_R1_CAMEL_VERSION = "2.18.1.redhat-000012";
-	public static final String FIS_20_R2_CAMEL_VERSION = "2.18.1.redhat-000015";
-	public static final String FIS_20_R3_CAMEL_VERSION = "2.18.1.redhat-000021";
-	
-	public static final String DEFAULT_CAMEL_VERSION = FIS_20_R3_CAMEL_VERSION;
+	public static final String DEFAULT_CAMEL_VERSION = CamelForFIS6ToBomMapper.FIS_20_R3_CAMEL_VERSION;
```
Comment (lhein) [inline core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/util/CamelCatalogUtils.java]: that assignment looks a bit weird at first sight. but I guess thats fine for now.

Comment (apupier) [reply core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/util/CamelCatalogUtils.java]: What do you find weird in it? That it is taking a value from CamelForFIS6ToBomMapper ? or that it is the FIS 2.0 R3?

File: core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/util/CamelCatalogUtils.java (hunk 1 above)
Comment (apupier) [reply core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/util/CamelCatalogUtils.java]: no big reasons except that it was to make clear opposition with "Fuse 7 On OpenShift". FIS 2.0 seems not clear enough when i wrote this classname but maybe better to keep FIS 2.0 to better match other naming.

File: core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/util/CamelCatalogUtils.java (hunk 2 above)
Comment (lhein) [reply core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/util/CamelCatalogUtils.java]: actually both :D

Comment (lhein) [reply core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/util/CamelCatalogUtils.java]: wondering if we should define that value directly in that class

Comment (apupier) [reply core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/util/CamelCatalogUtils.java]: I prefer to let it here for now. The default Camel version is "cross-Mappers".

**Pull request merged**